SubmittingPatches.rst

```diff
@@ -62,468 +66,207 @@ then you just add a line saying ::
+     * fix a typo
 
-     mds: add perf counter for finisher of MDSRank
+     Fixes: http://tracker.ceph.com/issues/12345
+     Signed-off-by: Random J Developer <random@developer.example.org>
 
-or::
+If a commit fixes a regression introduced by a different commit, please
+add a line referencing the SHA1 of the commit that introduced the regression.
+For example::
 
-     osd: make the ClassHandler::mutex private
+     Fixes: 9dbe7a003989f8bb45fe14aaa587e9d60a392727
 
-For GitHub, please use this "subsystem: short description" convention for
-naming pull requests (PRs). These titles feed directly into the script that
-generates release notes and it is tedious to clean up non-conformant PR titles
-at release time.  This document places no limit on the length of PR titles, but
-be aware that they are subject to editing as part of the release process.
 
-Commit message
-^^^^^^^^^^^^^^
+.. _`commit message`:
 
-Be as specific as possible. The WORST descriptions possible include
-things like "update driver X", "bug fix for driver X", or "this patch
-includes updates for subsystem X. Please apply."
+Writing commit messages
+-----------------------
 
-If your description starts to get long, that's a sign that you probably
-need to split up your patch. See :ref:`split_changes`.
+(This section is about the body of the commit message. Please also see
+the `Title of commit`_ section for advice on titling commit messages.)
 
-When you submit or resubmit a patch or patch series, include the
-complete patch description and justification for it. Don't just
-say that this is version N of the patch (series). Don't expect the
-patch merger to refer back to earlier patch versions or referenced
-URLs to find the patch description and put that into the patch.
-I.e., the patch (series) and its description should be self-contained.
-This benefits both the patch merger(s) and reviewers. Some reviewers
-probably didn't even receive earlier versions of the patch.
+In the body of your commit message, be as specific as possible. The WORST
+descriptions possible include things like "update component X", "bug fix for
+component X", or "this patch includes updates for subsystem X.  Please apply."
 
-Tag the commit
-^^^^^^^^^^^^^^
+There's nothing inherently wrong with a long commit message body. But it might
+be a sign that the patch could benefit from being split up into smaller patches.
+See the `Separate your changes`_ section.
 
-If the patch fixes a logged bug entry, refer to that bug entry by
-URL. In particular, if this patch fixes one or more issues
-tracked by http://tracker.ceph.com, consider adding a ``Fixes:`` tag to
-connect this change to addressed issue(s). So a line saying ::
+When you open a GitHub PR with multiple patches, the patches should all be
+related in some way and the PR title should state briefly what the patches do
+and why they are needed. In other words, the patch series and its description
```

Comment:
i don't think the title should state the "why" part. the body of a commit message should.